lang/wasm.cc

```diff
@@ -227,8 +235,12 @@ void precompile(context& ctx, const std::vector<sstring>& arg_names, std::string
         create_func(*inst, *store, ctx.function_name);
         ctx.module.value()->release();
     } catch (const rust::Error& e) {
-        throw wasm::exception(e.what());
+        ex = std::make_exception_ptr(wasm::exception(format("Compilation failed: {}", e.what())));
+    }
+    if (ex) {
+        co_await coroutine::return_exception_ptr(std::move(ex));
     }
+    co_return;
```

Comment:
nit, no need to call `co_return` explicitly here.

Reply:
Ok